packages/liveblocks-core/src/connection.ts

```diff
@@ -0,0 +1,756 @@
+  | "authenticating" // Authentication has started, but not finished yet
+  | "connecting" // Authentication succeeded, now attempting to connect to a room
+  | "open" // Successful room connection, on the happy path
+  | "unavailable" // Connection lost unexpectedly, considered a temporary hiccup, will retry
+  | "failed"; // Connection failed and we won't retry automatically (e.g. unauthorized)
+
+/**
+ * Maps internal machine state to the public connection status API.
+ */
+function toPublicConnectionStatus(state: State): PublicConnectionStatus {
+  switch (state) {
+    case "@ok.connected":
+    case "@ok.awaiting-pong":
+      return "open";
+
+    case "@idle.initial":
+      return "closed";
+
+    case "@auth.busy":
+    case "@auth.backoff":
+      return "authenticating";
+
+    case "@connecting.busy":
+      return "connecting";
+
+    case "@connecting.backoff":
+      return "unavailable";
+
+    case "@idle.failed":
+      return "failed";
+
+    default:
+      return assertNever(state, "Unknown state");
+  }
+}
+
+/**
+ * Events that can be sent to the machine externally.
+ */
+type Event =
+  // Public events that can be called on the connection manager
+  | { type: "CONNECT" } // e.g. when trying to enter a room
+  | { type: "RECONNECT" } // e.g. user asking for an explicit reconnect of the socket
+  | { type: "DISCONNECT" } // e.g. leaving the room
+  | { type: "WINDOW_GOT_FOCUS" } // e.g. user's browser tab is refocused
+  | { type: "NAVIGATOR_ONLINE" } // e.g. browser gets back online
+
+  // Events that the connection manager will internally deal with
+  | { type: "PONG" }
+  | { type: "EXPLICIT_SOCKET_ERROR"; event: IWebSocketEvent }
+  | { type: "EXPLICIT_SOCKET_CLOSE"; event: IWebSocketCloseEvent }
+
+  // Only used by the E2E testing app, to simulate a pong timeout :(
+  | { type: "PONG_TIMEOUT" };
+
+type State =
+  | "@idle.initial"
+  | "@idle.failed"
+  | "@auth.busy"
+  | "@auth.backoff"
+  | "@connecting.busy"
+  | "@connecting.backoff"
+  | "@ok.connected"
+  | "@ok.awaiting-pong";
+
+/**
+ * Arbitrary record that will be used as the authentication "token". It's the
+ * value that is returned by calling the authentication delegate, and will get
+ * passed to the connection factory delegate. This value will be remembered by
+ * the connection manager, but its value will not be interpreted, so it can be
+ * any object value.
+ */
+export type BaseAuthResult = Record<string, unknown>;
+
+type Context = {
+  /**
+   * Will be populated with the last known auth token.
+   */
+  token: BaseAuthResult | null;
+
+  /**
+   * The current active WebSocket connection to the room. If this is not null
+   * on the context, then the socket has successfully been opened.
+   */
+  socket: IWebSocketInstance | null;
+
+  /**
+   * The current retry delay when automatically retrying.
+   */
+  backoffDelay: number;
+  // numRetries: number; // Keep track of the number of retry attempts
```

Comment:
seems like we're using the backoff delays array, so you can just delete numRetries

Reply:
Agreed, fixed in e8445a96b149ea4f5bd71a3e0c7d8b12e67ac048 👍